[Bug] No longer able to nest 3 levels deep

### Is this a regression?

Yes

### Description

Prior to version 20, I have been able to nest the side bar 3 levels deep.  ie

```
+- Item
+- Item
  +-Sub Item
    - Sub-Sub-Item << no longer works
```

I notice the documentation for ng-doc only nest 2 levels.

I was able to narrow the problem to the third nested item by moving my documentation out to a backup folder and putting things back until I got to one that had items nested at the third level when things blew up.  I also verified that it only took one sub-sub-item to break the documentation build and there was nothing strange about it.

Link to my repo https://github.com/DaveMBush/SmartNgRX upgraded to version 20

I'm stuck because version 19 doesn't work with Angular 20 and version 20 won't build my documentation

### Please provide the exception or error you saw

```true
✘ [ERROR] An error occurred while extracting routes.

entryaulufyl1wwc is not defined



———————————————————————————————————————

 NX   Ran target build for project documentation and 1 task(s) they depend on (27s)

   ✖  1/1 failed
   ✔  0/1 succeeded [0 read from cache]
```

### OS

Unix (Linux, macOS, etc.)

### Browser

None

### Node version

22.16.0

### Anything else?

_No response_

--- a/libs/builder/templates/routes.ts.nunj
+++ b/libs/builder/templates/routes.ts.nunj
@@ -1,13 +1,16 @@
 // @ts-nocheck
 import {Routes} from '@angular/router';
 {%- macro categoryImport(entry) -%}
-{%- if entry.isCategory %}
-import {{ entry.id }} from '{{ curDir | createImportPath(entry.path) }}';
+{%- if entry.item.isCategory %}
+import {{ entry.item.id }} from '{{ curDir | createImportPath(entry.item.path) }}';
 {%- endif -%}
+{%- for child in entry.children -%}
+{{ categoryImport(child) }}
+{%- endfor -%}
 {%- endmacro -%}
 
 {%- for entry in entries -%}
-{{ categoryImport(entry.item) }}
+{{ categoryImport(entry) }}
 {%- endfor %}
 
 export const NG_DOC_ROUTING: Routes = [

